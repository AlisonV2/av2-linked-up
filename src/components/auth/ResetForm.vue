<template>
  <form class="auth-form" @submit.prevent="resetPassword">
    <div class="row">
      <div class="col-12 text-center mb-4">
        <h2 class="title">Reset Password</h2>
      </div>
    </div>
    <div class="form-floating mb-3">
      <input
        type="email"
        class="form-control"
        placeholder="Please enter your email"
        v-model="email"
        required
      />
      <label>Email</label>
    </div>
    <div class="error mb-3 mt-3">{{ error }}</div>
    <div class="btn-center">
      <app-button>Reset password</app-button>
    </div>
    <slot />
  </form>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports ParticipationForm
 * @type {Component}
 * @requires Sentry
 * @vue-data {string} email - v-model
 * @vue-event resetPassword
 */
export default {
  data() {
    return {
      email: '',
      error: null,
    };
  },
  methods: {
    resetPassword() {
      this.$store
        .dispatch('resetPassword', this.email)
        .then(() => {
          this.$router.push({ name: 'Login' });
        })
        .catch((err) => {
          this.error = err.message;
          Sentry.captureException(err);
          return;
        });
    },
  },
};
</script>

<style lang="scss" scoped>
.title {
  color: $dark;
  font-size: 1.8rem;
  margin-bottom: 1.5rem;
}
</style>
