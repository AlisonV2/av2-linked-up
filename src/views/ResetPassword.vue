<template>
  <div class="container col-6 home">
    <div>
      <ResetForm />
    </div>
  </div>
</template>

<script>
import ResetForm from '@/components/auth/ResetForm';

/**
 * @exports ResetPassword
 * @requires ResetForm
 * @type {Page}
 */
export default {
  name: 'ResetPassword',
  components: {
    ResetForm,
  },
};
</script>
