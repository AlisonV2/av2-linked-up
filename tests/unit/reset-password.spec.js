import { shallowMount } from '@vue/test-utils';
import ResetPassword from '@/views/ResetPassword.vue';
import ResetForm from '@/components/auth/ResetForm';

const config = {
  global: {
    components: {
      ResetForm,
    },
  },
};
/**
 * @module ResetPasswordPageTest
 */
describe('ResetPassword.vue', () => {
  it('ResetPasswordPage Snapshot', () => {
    const component = shallowMount(ResetPassword, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check if ResetForm is rendered', () => {
    const component = shallowMount(ResetPassword, config);
    const geoloc = component.findComponent(ResetForm);
    expect(geoloc.exists()).toBe(true);
  });

});
