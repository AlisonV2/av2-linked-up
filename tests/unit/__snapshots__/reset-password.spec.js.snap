// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword.vue ResetPasswordPage Snapshot 1`] = `
<div
  class="container col-6 home"
>
  <div>
    <reset-form-stub />
  </div>
</div>
`;
